src/esp/sim/BatchReplayRenderer.cpp

```diff
@@ -106,6 +108,49 @@ BatchReplayRenderer::BatchReplayRenderer(
           Mn::Matrix4::from(rotation.toMatrix(), translation);
     }
 
+    void changeLightSetup(const esp::gfx::LightSetup& lights) override {
+      if (!renderer_.maxLightCount()) {
+        ESP_WARNING() << "Attempted to change" << lights.size()
+                      << "lights for scene" << sceneId_
+                      << "but the renderer is configured without lights";
+        return;
+      }
+
+      renderer_.clearLights(sceneId_);
+      for (std::size_t i = 0; i != lights.size(); ++i) {
+        const gfx::LightInfo& light = lights[i];
+        CORRADE_INTERNAL_ASSERT(light.model == gfx::LightPositionModel::Global);
+
+        const std::size_t nodeId = renderer_.addEmptyNode(sceneId_);
+
+        /* Clang Tidy, you're stupid, why do you say that "lightId" is not
+           initialized?! I'm initializing it right in the branches below, I
+           won't zero-init it just to "prevent bugs" because an accidentally
+           zero-initialized variable is *also* a bug, you know? Plus I have
+           range asserts in all functions so your "suggestions" are completely
+           unhelpful. */
+        std::size_t lightId;  // NOLINT
+        if (light.vector.z()) {
```

Comment:
We want to look at the `w` component of the light vector to determine its type.

Reply:
@0mdc I am not following, is this flagging a bug as it is currently written?

Reply:
Oh, whoops -- yep, that's a bug. Fixing...

Reply:
Fixed in 3906aaad2199121ba8d2928ab025828c0d6b3170.